fontique/src/font.rs

```diff
@@ -430,14 +430,16 @@ fn read_attributes(font: &FontRef<'_>) -> (FontWidth, FontStyle, FontWeight) {
 
     fn from_head(head: Head<'_>) -> (FontWidth, FontStyle, FontWeight) {
         let mac_style = head.mac_style();
-        let style = mac_style
-            .contains(MacStyle::ITALIC)
-            .then_some(FontStyle::Italic)
-            .unwrap_or_default();
-        let weight = mac_style
-            .contains(MacStyle::BOLD)
-            .then_some(700.0)
-            .unwrap_or_default();
+        let style = if mac_style.contains(MacStyle::ITALIC) {
+            FontStyle::Italic
+        } else {
+            Default::default()
+        };
+        let weight = if mac_style.contains(MacStyle::BOLD) {
+            700.0
+        } else {
+            Default::default()
```

Comment:
```suggestion
            f32::default()
```
Which then pretty immediately falls through to

```suggestion
            0.0
```
Which doesn't seem right as a weight...
I'd suggest these branches should use `FontWeight::BOLD` and `FontWeight::NORMAL`, but I'm not sure.
I realise that would be a change of semantics, and so feel free to defer this (but if you do so and my understanding that this is suspect is correct, move this comment out into its own issue).

Either way, this lint has helped discover this.

Reply:
Agree. That looks like a bug to me. I think I'll fix it in a separate PR though so it isn't sneaked into a PR that looks like a form-only change.